### PR TITLE
refs #46: extended animation API for characters

### PR DESCRIPTION
**Pull request by stickgrinder**
This simple one introduces three useful functions.

### CODE REVIEW

File: addons/popochiu/engine/objects/character/popochiu_character.gd [hunk 1]
```diff
@@ -346,6 +358,25 @@ func play_animation(animation_label: String, animation_fallback := 'idle'):
...
 
+func stop_animation():
+	# TODO: consider returning if idle is playing, but in a way
+	# that's directions and animation_prefix aware!
+	var animation = $AnimationPlayer.get_animation($AnimationPlayer.current_animation)
+	var animation_loop_mode = animation.loop_mode
+	animation.loop_mode = Animation.LOOP_NONE
+	await $AnimationPlayer.animation_finished
+	_play_idle()
+	animation.loop_mode = animation_loop_mode
```
Comment (mapedorr) [inline addons/popochiu/engine/objects/character/popochiu_character.gd:369]: What would happen if one wants to stop an animation that is not on loop? Shouldn't this function call $AnimationPlayer.stop()?

**Review by mapedorr — COMMENTED** [review]
I just have a question about the `stop()` function. Otherwise this is ready for merge.

File: addons/popochiu/engine/objects/character/popochiu_character.gd (hunk 1 above)
Comment (stickgrinder) [reply addons/popochiu/engine/objects/character/popochiu_character.gd:369]: I was waiting for this question :D

I intended this function to stop the animation organically, when a cycle is over, going back to the idle state. This does not abruptly stop an animation midway through. If a long non-looping animation is running, this function basically does nothing.

Now that I think of it I can just test this case...

Comment (stickgrinder) [reply addons/popochiu/engine/objects/character/popochiu_character.gd:369]: One more thing is, we can parametrize this function to extend its behavior to stop the animation suddenly, no matter the state.

**Pull request merged**